### PR TITLE
fix: fixing pagination onchange with updated state values

### PR DESCRIPTION
**Pull request by Amarjit-Singh-22**
Issue :- The state passed in the onChange of the Pagination Component was the previous state. To make sure the onChange runs only when the state has been updated I have made these changes. (Done)

Snapshot of passing all the test cases

Screenshot 2023-01-30 at 11 49 01 AM

### CODE REVIEW

**Review by aadeshkulkarni-fynd — APPROVED** [review]

File: src/components/Pagination/Pagination.tsx [hunk 1]
```diff
@@ -48,9 +48,20 @@ const Pagination = (props: PaginationProps) => {
   const [selectedPageSize, setSelectedPageSize] = useState<any>(
     pageSizeOptions && pageSizeOptions.length > 0 ? pageSizeOptions[0] : 10
   );
+  const isFirstRender = useRef<boolean>(true);
+
   useEffect(() => {
     setDefaults();
   }, []);
+
+  useEffect(() => {
```
Comment (Zissan) [inline src/components/Pagination/Pagination.tsx:57]: Please add test cases to cover the newly added code

**Comment by Zissan** [comment]
Looks good to me

**Comment by Zissan** [comment]
I would recommend the snapshot of the PR to be added.

**Review by Zissan — APPROVED** [review]

**Comment by ShahRohan27** [comment]
@Amarjit-Singh-22 Please add a snapshot of the test case coverage as well.

**Pull request merged**